lib/pbench/server/api/resources/query_apis/query_result.py

```diff
@@ -0,0 +1,257 @@
+        # information in Redis or postgreSQL probably makes sense; maybe
+        # we just hardcode until then?
+        uri_fragment = gen_month_range(self.prefix, ".v6.run-data.", start, end)
+
+        uri = f"{self.es_url}/{uri_fragment}/_search"
+        try:
+            # query Elasticsearch
+            es_response = requests.post(
+                uri,
+                params={"ignore_unavailable": "true"},
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+                json=payload,
+            )
+            es_response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            self.logger.exception("HTTP error {} from Elasticsearch post request", e)
+            abort(502, message="INTERNAL ERROR")
+        except requests.exceptions.ConnectionError:
+            self.logger.exception(
+                "Connection refused during the Elasticsearch post request"
+            )
+            abort(502, message="Network problem, could not post to Elasticsearch")
+        except requests.exceptions.Timeout:
+            self.logger.exception(
+                "Connection timed out during the Elasticsearch post request"
+            )
+            abort(504, message="Connection timed out, could not post to Elasticsearch")
+        except requests.exceptions.InvalidURL:
+            self.logger.exception(
+                "Invalid url {} during the Elasticsearch post request", uri
+            )
+            abort(500, message="INTERNAL ERROR")
+        except Exception as e:
+            self.logger.exception(
+                "Exception {!r} occurred during the Elasticsearch post request", e
+            )
+            abort(500, message="INTERNAL ERROR")
+        else:
+            run_metadata = {}
+            try:
+                es_json = es_response.json()
+                hits = es_json["hits"]["hits"]
+
+                # NOTE: we're expecting just one. We're matching by just the
+                # dataset name, which ought to be unique.
+                self.logger.info("{} datasets found", len(hits))
+                src = hits[0]["_source"]
+
+                # We're merging the "run" and "@metadata" sub-documents into
+                # one dictionary, and then tacking on the host tools info in
+                # its original form.
+                run_metadata.update(src["run"])
+                run_metadata.update(src["@metadata"])
+                result = {
+                    "runMetadata": run_metadata,
+                    "hostTools": src["host_tools_info"],
+                }
+            except KeyError:
+                self.logger.exception("ES response not formatted as expected")
+                abort(500, message="INTERNAL ERROR")
+            else:
+                # construct response object
+                return jsonify(result)
```

Comment:
> it would probably be better to move the return statement inside the try block, since that's the scope of result.

Well, OK, I s'pose the contents of the `else` clause are arguably part of the `try` scope.  And, moving the `return` inside the `try` block risks an exception raised by the `jsonify()` call being misconstrued.  But, I still think it would be better to get rid of the `else` somehow.  ;-)

Reply:
Yes, I know you hate `try/else`, but I've gotten used to it; and I do think it's useful here to keep the result access clearly "within scope" but outside the REST call failure domain.